router/src/metrics.rs

```diff
@@ -0,0 +1,43 @@
+// Small helpers for using the metrics crate.
+// This aims to collect all usages of the metrics crate so that future api-breaking changes can be handled in one place.
+
+
+// These counter helper methods will actually increment a second counter with `_total` appended to the name.
+// This is for compatibility with other runtimes that use prometheus directly, which is very
+// opinionated that all counters should end with the suffix _total.
+// Cite: https://prometheus.github.io/client_python/instrumenting/counter/
+
+pub fn increment_counter(name: &'static str, value: u64) {
```

Comment:
I did separate functions for labeled vs. not labeled metrics so we wouldn't have to add a default empty slice everywhere. IIUC Rust doesn't have default values for function arguments, or dynamic overloading with different argument list lengths, but of course I could be way wrong.